python/tvm/relay/frontend/pytorch.py

```diff
@@ -1546,6 +1546,16 @@ def flatten(self, inputs, input_types):
             out = _op.squeeze(out, axis=squeeze_axes)
         return out
 
+    def unflatten(self, inputs, input_types):
+        data = inputs[0]
+        dim = int(inputs[1])
+        unflattened_size = tuple(inputs[2])
+        dshape = get_const_tuple(self.infer_shape_with_prelude(data))
+        assert len(dshape) > dim
+        new_shape = dshape[:dim] + unflattened_size + dshape[dim + 1 :]
```

Comment:
Add check that dshape[dim] == multiplication of dimensions in unflattened_size

Reply:
@vvchernov Thank you for your comment!
Multiplication of dimensions in `unflattened_size` doesn't always match `dshape[dim]` because `unflattened_size` can contain -1.
How do we check it? Do you have any idea?

Reply:
@vvchernov Thanks!
I don't think we have to check it because torch.jit.trace does it.
They provide something like the below RuntimeError.
```
RuntimeError: unflatten: Provided sizes [3, 5, 3, -1] don't multiply up to the size of dim 0 (60) in the input tensor
```
Should we add the check in TVM's PyTorch frontend?

Reply:
Hello @mshr-h! Ok torch.jit.trace do it, but in this case we do not need `assert len(dshape) > dim`.
It looks like TVM usually rechecks all corner cases.
About -1 in unflattened_size. In this case we can multiply together other dimension and check dshape[dim] % mult == 0

Reply:
@vvchernov Thanks. Added the assertion.

Reply:
Hello @mshr-h! There are two cases: unflattened_size has -1 and does not have. You check only the first one. Example:  dshape[dim] = 8, unflattened_size = [2, 2, 1, 1] pass your assert, but it is failure case

Reply:
I've checked that _op.reshape does not check size correctness by it-self. Nevertheless I found that dim can be not only -1, but from list {0, -1, -2, -3, -4} See https://github.com/apache/tvm/blob/748882aae7be1435f042e22b0fc67cb236705b6c/python/tvm/relay/op/transform.py#L243
I suggest to check -1 case only. It looks like torch does not have other options

Reply:
@vvchernov 
Thanks! Updated the shape check.

Reply:
Ah, dim can be negative but I didn't check that. I'll add the check.

Reply:
> Ah, dim can be negative but I didn't check that. I'll add the check.

Done.